xbmc/addons/AddonEvents.h

```diff
@@ -54,34 +62,31 @@ namespace ADDON
      */
     struct ReInstalled: AddonEvent
     {
-      std::string id;
-      explicit ReInstalled(std::string id) : id(std::move(id)) {}
+      explicit ReInstalled(std::string id) : AddonEvent(std::move(id)) {}
     };
 
     /**
      * Emitted after the add-on has been uninstalled.
      */
     struct UnInstalled : AddonEvent
     {
-      std::string id;
-      explicit UnInstalled(std::string id) : id(std::move(id)) {}
+      explicit UnInstalled(std::string id) : AddonEvent(std::move(id)) {}
     };
 
     /**
-     * @deprecated Use Enabled, ReInstalled and UnInstalled instead.
+     * Emitted after the add-on has been loaded.
      */
-    struct InstalledChanged : AddonEvent {};
-
     struct Load : AddonEvent
     {
-      std::string id;
-      explicit Load(std::string id) : id(std::move(id)) {}
+      explicit Load(std::string id) : AddonEvent(std::move(id)) {}
     };
 
+    /**
+     * Emitted after the add-on has been unloaded.
```

Comment:
This is wrong. Both Load and Unload are before (that's the entire point). Actually, they should not even be part of AddonEvent. That's just more stuff from #12473 that wasn't completed.